darts/tests/test_timeseries.py

```diff
@@ -2202,6 +2206,28 @@ def test_time_col_convert_garbage(self):
         with self.assertRaises(AttributeError):
             TimeSeries.from_dataframe(df=df, time_col="Time")
 
+    def test_df_named_columns_index(self):
+        time_index = generate_index(
+            start=pd.Timestamp("2000-01-01"), length=4, freq="D", name="index"
+        )
+        df = pd.DataFrame(
+            data=np.arange(4),
+            index=time_index,
+            columns=["y"],
+        ).reset_index()
+        df["id"] = list("AB") * 2
+        # df.columns.name = "id"
+        df = df.pivot(columns="id", values="y", index="index").reset_index()
+
+        ts = TimeSeries.from_dataframe(df, time_col="index", fillna_value=-1)
+
+        exp_ts = TimeSeries.from_times_and_values(
+            times=time_index,
+            values=np.array([[0, -1], [-1, 1], [2, -1], [-1, 3]]),
+            columns=pd.Index(["A", "B"]),
+        )
+        self.assertEqual(ts, exp_ts)
```

Comment:
Wouldn't it be enough just to set the name for the column index?
```suggestion
        df = pd.DataFrame(
            data=np.arange(4),
            index=time_index,
            columns=["y"],
        )
        df.columns.name = "id"
        ts = TimeSeries.from_dataframe(df)

        exp_ts = TimeSeries.from_times_and_values(
            times=time_index,
            values=np.arange(4),
            columns=["y"],
        )
        self.assertEqual(ts, exp_ts)
```